app/views/public/babel_view.py

```diff
@@ -0,0 +1,9 @@
+from flask import request, Blueprint, session
+
+app = Blueprint('babel', __name__)
+
+@app.route('/choose_language', methods=('POST',))
+def set_lang():
+    l_code = request.form.get('l_code')
+    session["selected_lang"] = l_code
```

Comment:
Don't store it in a session. A session is temporary. If the user closes his browser and open the site again, he'll not get his preferred language. So, as I said, use a cookie. (Ref: http://flask.pocoo.org/snippets/30/)